peer_mentoring/groups/views.py

```diff
@@ -1,7 +1,61 @@
-from django.shortcuts import render
+from account_management.models import UserProfile
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import get_object_or_404, redirect, render
+from posts.models import Comment, Post
+
+from .forms import GroupPostCommentForm, GroupPostForm
+from .models import Group
 
 # Create your views here.
 
 
-def group_view(request):
-    pass
+@login_required
+def group_index_view(request):
+    groups = Group.objects.all()
+    context = {"groups": groups}
+    return render(request, "groups_index.html", context)
+
+
+@login_required
+def group_detail_view(request):
+    group = get_object_or_404(Group, pk=Group.id)
+    context = {
+        "group": group,
+    }
+    return render(request, "group_detail.html", context)
+
+
+@login_required
+def group_posts(request):
+    posts = Post.objects.all().filter(id=Group.id)
+    comments = request.POST.get(Comment)
```

Comment:
does this work? I would not expect a Comment object in the POST dict

Reply:
I haven't tested, don't have template up. However, how would you add a comment to the posts in a view?

Reply:
I would use my comment model form with commit=False to manually add the post object onto it:

```
form = CommentForm(request.POST)
if form.is_valid():
    comment = form.save(commit=False
    comment.post = post  # the retrieved post object
    comment.save()
```

Reply:
Can you please explain this a little further tomorrow? As in, what is the logic doing here? It's saving but he commit=False doesn't save to db. The post method posts the comment object to the post object, then saves it to the post object? I know this seems confusing but that's how I am making sense of it right now.